admin/src/containers/Rectifications/RectificationsContext.js

```diff
@@ -0,0 +1,122 @@
+import React, { Component, createContext } from 'react';
+import PropTypes from 'prop-types';
+import config from 'config';
+import internalFetch from 'helpers/internal-fetch';
+import { toast } from 'react-toastify';
+import { format } from 'date-fns';
+
+const RectificationsContext = createContext({
+  pendingRectifications: {},
+  processedRectifications: {},
+  fetchPendingRectifications: () => {},
+  fetchProcessedRectifications: () => {},
+  fetchAllRectifications: () => {},
+  isLoading: false
+});
+
+export class RectificationsProvider extends Component {
+  static propTypes = {
+    children: PropTypes.oneOfType([
+      PropTypes.element,
+      PropTypes.object,
+      PropTypes.arrayOf(PropTypes.element)
+    ])
+  };
+
+  state = {
+    pendingRectifications: {},
+    processedRectifications: {},
+    fetchPendingRectifications: this.fetchPendingRectifications.bind(this),
+    fetchProcessedRectifications: this.fetchProcessedRectifications.bind(this),
+    fetchAllRectifications: this.fetchAllRectifications.bind(this),
+    isLoading: false
+  };
+
+  async _getPendingRectifications(page = 1) {
+    return this._mapRectificationResults(
+      await internalFetch(
+        `${config.API_URL}/api/management/subjects/rectification-requests/list?page=${page}`
+      )
+    );
+  }
+
+  async _getProcessedRectifications(page = 1) {
+    return this._mapRectificationResults(
+      await internalFetch(
+        `${config.API_URL}/api/management/subjects/rectification-requests/archive?page=${page}`
+      )
+    );
+  }
+
+  _mapRectificationResults(result) {
+    let mappedData = result.data.map(v => {
+      let mapped = {
+        id: v.id,
+        created_at: format(new Date(v.created_at), 'DD/MM/YYYY h:mma'),
+        request_reason: v.request_reason
+      };
+
+      v.status && (mapped.status = v.status);
```

Comment:
I would prefer this written as `if(v.status) mapped.status = v.status` for readability